implementations/rust/ockam/ockam_api/src/nodes/models/credentials.rs

```diff
@@ -0,0 +1,70 @@
+//! Credentials request/response types
+
+use minicbor::{Decode, Encode};
+use ockam_core::compat::borrow::Cow;
+use ockam_core::CowStr;
+
+#[cfg(feature = "tag")]
+use ockam_core::TypeTag;
+use ockam_multiaddr::MultiAddr;
+
+#[derive(Clone, Debug, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct SetAuthorityRequest {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<1864428>,
+    #[b(1)] pub authorities: Vec<Vec<u8>>,
+}
+
+impl SetAuthorityRequest {
+    pub fn new(authorities: Vec<Vec<u8>>) -> Self {
+        Self {
+            #[cfg(feature = "tag")]
+            tag: TypeTag,
+            authorities,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct GetCredentialRequest<'a> {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<8479533>,
+    #[b(1)] pub route: CowStr<'a>,
+    #[n(2)] pub overwrite: bool,
+}
+
+impl<'a> GetCredentialRequest<'a> {
+    pub fn new(route: &MultiAddr, overwrite: bool) -> Self {
+        Self {
+            #[cfg(feature = "tag")]
+            tag: TypeTag,
+            route: route.to_string().into(),
```

Comment:
`route : MultiAddr` should work now.

Reply:
You mean having MultiAddr field in the structure?
```
29 | #[derive(Clone, Debug, Decode, Encode)]
   |                        ^^^^^^ the trait `minicbor::Decode<'_, Ctx>` is not implemented for `MultiAddr`
```

Reply:
Yes, it requires the `cbor` feature of `ockam_multiaddr`.

Reply:
@twittner any reason this feature is not enabled by default?

Reply:
And too bad when cbor support was added it wasn't propagated to the places where it could be useful...

Reply:
I'll fix that...

Reply:
My opinion is that features should be opt-in, especially when they activate additional dependencies. You can change that of course.